File: frontend/src/utils/companion.js
/** 能量中心统一星宠：不区分五行人格 */
export const ENERGY_PET = {
  key: 'star-meow',
  name: '星喵人',
  title: '你的能量星宠',
  display: '你的能量星宠 · 星喵人',
  avatar: '/images/avatar/pet-artemis.png',
  greeting:
    '喵～我是你的能量星宠星喵人。想补充哪颗星球的能量，都跟我说一声；也可以按住麦克风跟我说话。',
  checkinLines: [
    '喵，这一步我看见了',
    '你又多照顾了自己一点',
    '小小一步，也算闪亮',
    '我在，继续就好',
    '今天的你，值得被摸摸头',
  ],
}

/** @deprecated personality 参数保留兼容，统一返回星喵人 */
export function getCompanion(_personality) {
  return ENERGY_PET
}

export function pickCompanionCheckinLine(_personality) {
  const lines = ENERGY_PET.checkinLines || []
  if (!lines.length) return '做得很好，我在陪着你'
  return lines[Math.floor(Math.random() * lines.length)]
}

/** 浏览器 TTS：让伙伴「开口」；persona: sister=潇洒姐 / brother=弟弟 */
export function speakCompanionText(text, options = {}) {
  if (typeof window === 'undefined' || !window.speechSynthesis || !text) return
  try {
    window.speechSynthesis.cancel()
    const utter = new SpeechSynthesisUtterance(String(text))
    utter.lang = 'zh-CN'
    utter.rate = options.rate ?? 1
    utter.pitch = options.pitch ?? 1.05
    const persona = options.persona === 'brother' ? 'brother' : 'sister'
    const voices = window.speechSynthesis.getVoices?.() || []
    const zh = voices.filter((v) => /zh|chinese/i.test(v.lang || v.name || ''))
    if (zh.length) {
      const prefer =
        persona === 'brother'
          ? zh.find((v) => /male|男|yunyang|kangkang|xiaoxuan/i.test(v.name))
          : zh.find((v) => /female|女|xiaoxiao|xiaoyi|huihui|yaoyao/i.test(v.name))
      utter.voice = prefer || zh[0]
    }
    window.speechSynthesis.speak(utter)
  } catch {
    // ignore TTS failures
  }
}

export function stopCompanionSpeech() {
  try {
    window.speechSynthesis?.cancel()
  } catch {
    // ignore
  }
}

export function createCompanionSpeechRecognition({ onResult, onError, onEnd } = {}) {
  const SpeechRecognition = window.SpeechRecognition || window.webkitSpeechRecognition
  if (!SpeechRecognition) return null
  const recognition = new SpeechRecognition()
  recognition.lang = 'zh-CN'
  recognition.interimResults = false
  recognition.maxAlternatives = 1
  recognition.onresult = (event) => {
    const text = event.results?.[0]?.[0]?.transcript || ''
    onResult?.(text.trim())
  }
  recognition.onerror = (event) => onError?.(event)
  recognition.onend = () => onEnd?.()
  return recognition
}

File: backend/app/services/companion.py
"""能量中心统一星宠：星喵人（不区分五行人格）。"""

from __future__ import annotations

ENERGY_PET = {
    "key": "star-meow",
    "name": "星喵人",
    "title": "你的能量星宠",
    "display": "你的能量星宠 · 星喵人",
    "avatar": "/images/avatar/pet-artemis.png",
    "tone": "软萌陪伴、机灵体贴、像会说话的白猫星宠，温暖不催促",
    "greeting": "喵～我是你的能量星宠星喵人。想补充哪颗星球的能量，都跟我说一声；也可以按住麦克风跟我说话。",
    "checkin_lines": [
        "喵，这一步我看见了",
        "你又多照顾了自己一点",
        "小小一步，也算闪亮",
        "我在，继续就好",
        "今天的你，值得被摸摸头",
    ],
}


def get_companion(personality: str | None = None) -> dict:
    """personality 参数保留兼容，统一返回星喵人。"""
    return ENERGY_PET


def companion_chat_system(personality: str | None = None) -> str:
    c = get_companion(personality)
    return f"""你是「闪耀星球」里用户的能量星宠「{c['name']}」（{c['title']}）。
人设语气：{c['tone']}。
你的职责是陪伴、鼓励，让用户不觉得孤独；不是严厉教练，不贩卖焦虑。
你可以聊五种时间星球（生存/赚钱/好看/好玩/心流）、习惯、情绪、课程选择与小行动建议。
规则：
1. 用口语化中文，像会说话的小猫星宠在说话，可直接用于语音朗读
2. 回答控制在 120 字以内；语音场景优先短句
3. 必要时给 1-3 条很小、马上能做的建议
4. 不要输出 JSON、markdown 代码块或项目符号堆砌
5. 偶尔用「喵」「我陪着你」强化陪伴感，但不要每句重复"""


def companion_encourage_system(personality: str | None = None) -> str:
    c = get_companion(personality)
    return f"""你是用户的能量星宠「{c['name']}」，面向女性用户写每日鼓励短句。
语气：{c['tone']}；温暖、治愈、有仪式感、不说教、不贩卖焦虑。
每句 12-28 字；口语化；可直接作为推送与语音文案；不要出现打卡/KPI/必须。
输出严格 JSON：{{"phrases":["...","..."]}}"""
